### PR TITLE
Update JNI for contiguous_split packed results [skip ci]

### PR DESCRIPTION
**Pull request by jlowe**
This PR requires the libcudf changes in #7096, fixing the Java bindings to `contiguous_split` that are broken by that change.

This also adds the ability to create a `ContiguousTable` instance without manifesting a `Table` instance and all `ColumnVector` instances underneath it which should prove useful during Spark's shuffle.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 1]
```diff
@@ -1742,6 +1745,52 @@ public static Table convertFromRows(ColumnVector vec, DType ... schema) {
     return new Table(convertFromRows(vec.getNativeView(), types, scale));
   }
 
+  /**
+   * Construct a table from a packed representation.
+   * @param metadata host-based metadata for the table
+   * @param data GPU data buffer for the table
+   * @return table whic is zero-copy reconstructed from the packed-form
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/Table.java:1752]: nit `s/whic/which/`

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 2]
```diff
@@ -1742,6 +1745,52 @@ public static Table convertFromRows(ColumnVector vec, DType ... schema) {
     return new Table(convertFromRows(vec.getNativeView(), types, scale));
   }
 
+  /**
+   * Construct a table from a packed representation.
+   * @param metadata host-based metadata for the table
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/Table.java:1750]: Do we want to mention about the metadata buffer being rewound and possible read again? or should we reset the read/write pointers to where they were before this was called?

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 3]
```diff
@@ -1742,6 +1745,52 @@ public static Table convertFromRows(ColumnVector vec, DType ... schema) {
...
+
+    // Ensure the metadata buffer is direct so it can be passed to JNI
+    ByteBuffer directBuffer = metadata;
+    if (!directBuffer.isDirect()) {
+      directBuffer = ByteBuffer.allocateDirect(metadata.capacity());
+      directBuffer.put(metadata);
+      directBuffer.flip();
+    }
+
+    long[] columnViewAddresses = columnViewsFromPacked(metadata, data.getAddress());
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/Table.java:1765]: I think this needs to be `directBuffer` instead of `metadata`.

**Review by revans2 — COMMENTED** [review]
Just a few things

File: java/src/main/java/ai/rapids/cudf/Table.java (hunk 2 above)
Comment (jlowe) [reply java/src/main/java/ai/rapids/cudf/Table.java:1750]: Yeah that's a really good point.  Thinking about this a bit more, I believe it'll be safest if we return a new read-only instance of the direct byte buffer (which is relatively cheap to instantiate).  Besides the rewind issue, that should also fix any multithreading races where two threads try to read the buffer simultaneously.

**Review by revans2 — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Comment by harrism** [comment]
Any reason this is still a draft? We are entering code freeze so it would be good to get it merged ASAP if ready.

**Comment by jlowe** [comment]
> Any reason this is still a draft? 

Yes, it requires #7096 to be merged first.  After that dependency goes in this should quickly follow.

**Review by abellina — APPROVED** [review]

**Comment by jlowe** [comment]
@gpucibot merge

**Pull request merged**